### PR TITLE
Functions for oldest to newest iterator

### LINKED ISSUE #2: Iterator direction
Is there some specific reason why you `.rev()` each half of the iterator separately?

Also, is there a way to get the other direction (oldest to newest) without having to `.rev()` the iterator each time it is used? Perhaps you could add an iterator like the following:

```
let (first_half, second_half) = self.data.split_at(self.insertion_index);
second_half.iter().chain(first_half.iter())
```

### PR DESCRIPTION
**Pull request by valmirjunior0088**
Closes #2.

I tried my best to follow your current philosophy for the project, but perhaps I might need some help writing the tests.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -248,6 +254,59 @@ impl<T> CircularQueue<T> {
...
+    /// queue.push(4);
+    ///
+    /// let mut iter = queue.incr_iter();
+    ///
+    /// assert_eq!(iter.next(), Some(&2));
+    /// assert_eq!(iter.next(), Some(&3));
+    /// assert_eq!(iter.next(), Some(&4));
+    /// ```
+    #[inline]
+    pub fn incr_iter(&mut self) -> IncrIter<T> {
```
Comment (YaLTeR) [inline src/lib.rs]:
```suggestion
    pub fn incr_iter(&self) -> IncrIter<T> {
```

**Review by YaLTeR — CHANGES_REQUESTED** [review]
All good apart from the change, but I'm not sure I like the name "incremental", it sounds like a different concept. That said, I can't quite come up with a good name either, so let's leave this as is.

**Comment by YaLTeR** [comment]
The CI failure is due to https://github.com/rust-lang/rust/issues/67408.

**Comment by valmirjunior0088** [comment]
Perhaps 'ascending' transmits a meaning that more closely resembles our objective.

Regarding the CI failure, I'm not exactly sure how I would go about fixing it.

**Comment by valmirjunior0088** [comment]
Small curiosity: you added the `#[inline]` annotation to every method of the structure, but not the `.push()` method. Is there some reason?

**Comment by YaLTeR** [comment]
> Small curiosity: you added the `#[inline]` annotation to every method of the structure, but not the `.push()` method. Is there some reason?

Not that I can recall. Probably forgot about it. ¯\\\_(ツ)\_/¯

**Comment by valmirjunior0088** [comment]
I've took the liberty of adding it.

Regarding the entire PR, I believe this is enough functionality. Can we merge?

**Comment by YaLTeR** [comment]
Yes, great! Do you mind squashing it into two commits, one adding the missing `#[inline]` on `push` and another adding the iterators?

**Comment by valmirjunior0088** [comment]
Done.

**Comment by YaLTeR** [comment]
Thanks!

**Pull request merged**